ambry-tools/src/main/java/com.github.ambry/clustermap/HelixVcrPopulateTool.java

```diff
@@ -0,0 +1,183 @@
+import org.apache.helix.tools.ClusterSetup;
+
+
+/**
+ * This tool provides function to create vcr cluster and update vcr cluster by referencing src cluster.
+ */
+public class HelixVcrPopulateTool {
+
+  private static String SEPARATOR = "/";
+  static List<String> ignoreResourceKeyWords = Arrays.asList("aggregation", "trigger", "stats");
+
+  public static void main(String args[]) {
+    OptionParser parser = new OptionParser();
+    OptionSpec createClusterOpt = parser.accepts("createCluster",
+        "Create resources in dest by copying from src to dest. --createCluster --dest destZkEndpoint/destClusterName");
+    OptionSpec updateClusterOpt = parser.accepts("updateCluster",
+        "Update resources in dest by copying from src to dest. --updateCluster"
+            + " --src srcZkEndpoint/srcClusterName --dest destZkEndpoint/destClusterName");
+    OptionSpec dryRunOpt = parser.accepts("dryRun", "Do dry run.");
+
+    ArgumentAcceptingOptionSpec<String> srcOpt =
+        parser.accepts("src").withRequiredArg().describedAs("src zk and cluster name").ofType(String.class);
+    ArgumentAcceptingOptionSpec<String> destOpt =
+        parser.accepts("dest").withRequiredArg().describedAs("dest zk and cluster name").ofType(String.class);
+
+    OptionSet options = parser.parse(args);
+
+    String destZkString = options.valueOf(destOpt).split(SEPARATOR)[0];
+    String destClusterName = options.valueOf(destOpt).split(SEPARATOR)[1];
+    if (!destClusterName.contains("VCR")) {
+      System.out.println("dest should be a VCR cluster.(VCR string should be included)");
+      return;
+    }
+
+    if (options.has(createClusterOpt)) {
+      System.out.println("Creating cluster: " + destClusterName);
+      createCluster(destZkString, destClusterName);
+    }
+
+    if (options.has(updateClusterOpt)) {
+      System.out.println("Updating cluster: " + destClusterName);
+      String srcZkString = options.valueOf(srcOpt).split(SEPARATOR)[0];
+      String srcClusterName = options.valueOf(srcOpt).split(SEPARATOR)[1];
+      boolean dryRun = options.has(dryRunOpt);
+      updateResourceAndPartition(srcZkString, srcClusterName, destZkString, destClusterName, dryRun);
+    }
+    System.out.println("Done.");
+  }
+
+  /**
+   * Create a helix cluster with given information.
+   * @param destZkString the cluster's zk string
+   * @param destClusterName the cluster's name
+   * @return false if cluster already exist, otherwise true.
+   */
+  static boolean createCluster(String destZkString, String destClusterName) {
+    HelixZkClient destZkClient =
+        DedicatedZkClientFactory.getInstance().buildZkClient(new HelixZkClient.ZkConnectionConfig(destZkString));
+    destZkClient.setZkSerializer(new ZNRecordSerializer());
+    HelixAdmin destAdmin = new ZKHelixAdmin(destZkClient);
+    if (destAdmin.getClusters().contains(destClusterName)) {
+      System.out.println("Create cluster failed. " + destClusterName + " already exist.");
```

Comment:
minor: how about change to this: `System.out.println("Failed to create cluster because " + destClusterName + " already exists.");`